### PR TITLE
Feature/remove should determine best chain set tip

### PR DESCRIPTION
**Pull request by eu321**

### CODE REVIEW

File: include/veriblock/blockchain/alt_block_tree.hpp [hunk 1]
```diff
@@ -68,6 +68,13 @@ struct AltTree : public BaseBlockTree<AltBlock> {
   //! may return false, if bootstrap block is invalid
   bool bootstrap(ValidationState& state);
 
+  /**
+   * Check if the blockchain is bootstrapped
+   *
+   * @return true if the blockchain is bootstrapped, false otherwise
+   */
+  bool isBootstrapped();
```
Comment (Warchant) [inline include/veriblock/blockchain/alt_block_tree.hpp]: Should be moved to base_block_tree

File: src/blockchain/alt_block_tree.cpp [hunk 2]
```diff
@@ -460,37 +463,47 @@ bool AltTree::addPayloads(AltTree::index_t& index,
...
-                      : "<empty>"));
-    activeChain_.setTip(&to);
-    tryAddTip(&to);
+  if (success) {
+    bool success = overrideTip(to, state);
+    VBK_ASSERT(success);
   } else {
+    // if setState failed, then 'to' must be invalid
     VBK_ASSERT(!to.isValid());
   }
```
Comment (Warchant) [inline src/blockchain/alt_block_tree.cpp:489]: no need in this condition, because `success` is always true

Comment (eu321) [reply src/blockchain/alt_block_tree.cpp:489]: setState is definitely not always going to succeed

Comment (Warchant) [reply src/blockchain/alt_block_tree.cpp:489]: look closely - this is setTipContinueOnInvalid, it always succeeds, and there are two asserts that indicate this
https://github.com/VeriBlock/alt-integration-cpp/pull/291/files#diff-2cf3b8cf328ee87b5b03c1b7d0abf949R481
https://github.com/VeriBlock/alt-integration-cpp/pull/291/files#diff-2cf3b8cf328ee87b5b03c1b7d0abf949R485

Comment (eu321) [reply src/blockchain/alt_block_tree.cpp:489]: ah right. github doesn't show enough context lines :(

**Pull request merged**